### PR TITLE
[SPARK-7217][STREAMING] Add configuration to control the default behavior of StreamingContext.stop() implicitly calling SparkContext.stop()

### PR DESCRIPTION
**Pull request by tdas**
In environments like notebooks, the SparkContext is managed by the underlying infrastructure and it is expected that the SparkContext will not be stopped. However, StreamingContext.stop() calls SparkContext.stop() as a non-intuitive side-effect. This PR adds a configuration in SparkConf that sets the default StreamingContext stop behavior. It should be such that the existing behavior does not change for existing users.

### CODE REVIEW

**Comment by tdas** [comment]
@pwendell @srowen Can you take a quick look at this. Any thought on the configuration name?

**Comment by harishreedharan** [comment]
LGTM

File: streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala [hunk 1]
```diff
@@ -563,13 +563,22 @@ class StreamingContext private[streaming] (
...
+  /**
+   * Stop the execution of the streams immediately (does not wait for all received data
    * to be processed).
    *
    * @param stopSparkContext if true, stops the associated SparkContext. The underlying SparkContext
    *                         will be stopped regardless of whether this StreamingContext has been
    *                         started.
    */
-  def stop(stopSparkContext: Boolean = true): Unit = synchronized {
+  def stop(stopSparkContext: Boolean): Unit = synchronized {
```
Comment (harishreedharan) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: Hmm, this is actually api incompatible change no? I am wondering if there is a workaround here which does not break compat.

Comment (tdas) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: I thought it wasnt, but it seems to be. I will implement it differently.

**Comment by srowen** [comment]
Seems OK to me. I agree with leaving it "hidden".

**Comment by tdas** [comment]
@srowen Yep, that's the plan for now.

**Comment by tdas** [comment]
I am merging this. Thanks @srowen @harishreedharan for taking a look.

**Pull request closed without merging**